### PR TITLE
Bug 1762217 - WebCompat Interventions v100.2.0

### PR DESCRIPTION
**Pull request by denschub**
Primarily a lot of Version 100 user agent overrides, and removing an Instagram intervention that's no longer needed.

r? @Amejia481

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/feature/webcompat/src/main/assets/extensions/webcompat/manifest.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "manifest_version": 2,
   "name": "Mozilla Android Components - Web Compatibility Interventions",
   "description": "Urgent post-release fixes for web compatibility.",
-  "version": "100.0.0",
+  "version": "100.2.0",
```
Comment (Amejia481) [inline components/feature/webcompat/src/main/assets/extensions/webcompat/manifest.json:5]: Just to confirm, that we are jumping from one to two intentionally? :)

**Review by Amejia481 — APPROVED** [review]
Looks good in overall just one question.

File: components/feature/webcompat/src/main/assets/extensions/webcompat/manifest.json (hunk 1 above)
Comment (denschub) [reply components/feature/webcompat/src/main/assets/extensions/webcompat/manifest.json:5]: Oh, sorry, I somehow missed the notification, @Amejia481.

Yeah, we landed `.1` in mozilla-central only as the changes were irrelevant for mobile, so this was on purpose. Thanks for checking! :)

Comment (Amejia481) [reply components/feature/webcompat/src/main/assets/extensions/webcompat/manifest.json:5]: No problem, thanks for the clarification :)

**Pull request merged**